File: backend/src/main/java/com/pharma/backend/entity/donhang/DonHang.java
package com.pharma.backend.entity.donhang;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "don_hang")
@Getter
@Setter
public class DonHang {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ma_don_hang")
    private Long maDonHang;

    @Column(name = "ma_khach_hang")
    private Long maKhachHang;

    @Column(name = "ma_dia_chi")
    private Long maDiaChi;

    @Column(name = "ma_voucher")
    private Long maVoucher;

    @Column(name = "ma_don_thuoc")
    private Long maDonThuoc;

    @Column(name = "ma_nhan_vien_xu_ly")
    private Long maNhanVienXuLy;

    @Column(name = "ma_duoc_si_duyet")
    private Long maDuocSiDuyet;

    @Column(name = "ngay_dat_hang", nullable = false)
    private LocalDateTime ngayDatHang;

    @Column(name = "loai_khach", length = 30)
    private String loaiKhach;

    @Column(name = "tong_tien_hang", nullable = false, precision = 12, scale = 2)
    private BigDecimal tongTienHang;

    @Column(name = "phi_giao_hang", precision = 12, scale = 2)
    private BigDecimal phiGiaoHang;

    @Column(name = "giam_gia", precision = 12, scale = 2)
    private BigDecimal giamGia;

    @Column(name = "tong_thanh_toan", nullable = false, precision = 12, scale = 2)
    private BigDecimal tongThanhToan;

    @Column(name = "phuong_thuc_thanh_toan", length = 50)
    private String phuongThucThanhToan;

    @Column(name = "trang_thai_thanh_toan", length = 30)
    private String trangThaiThanhToan;

    @Column(name = "trang_thai_don_hang", nullable = false, length = 30)
    private String trangThaiDonHang;

    @Column(name = "trang_thai_kiem_duyet", nullable = false, length = 30)
    private String trangThaiKiemDuyet;

    @Column(name = "ngay_kiem_duyet")
    private LocalDateTime ngayKiemDuyet;

    @Column(name = "ghi_chu_kiem_duyet", length = 255)
    private String ghiChuKiemDuyet;

    @Column(name = "ly_do_tu_choi_duyet", length = 255)
    private String lyDoTuChoiDuyet;

    @Column(name = "ghi_chu", length = 255)
    private String ghiChu;
}

File: frontend/src/features/don-hang/types/DonHang.ts
export type PhanTrangResponse<T> = {
  content: T[];
  page: number;
  size: number;
  totalElements: number;
  totalPages: number;
  first: boolean;
  last: boolean;
};

export type LoaiKhachHang = "CO_TAI_KHOAN" | "VANG_LAI";

export type PhuongThucThanhToan = "COD" | "ZALOPAY" | "TIEN_MAT";

export type TrangThaiThanhToan =
  | "CHUA_THANH_TOAN"
  | "DA_THANH_TOAN"
  | "THANH_TOAN_THAT_BAI"
  | "DA_HOAN_TIEN";

export type TrangThaiDonHang =
  | "CHO_XU_LY"
  | "DANG_XU_LY"
  | "DANG_GIAO"
  | "HOAN_THANH"
  | "DA_HUY";

export type TrangThaiKiemDuyetDonHang =
  | "KHONG_CAN_DUYET"
  | "CHO_DUYET"
  | "DA_DUYET"
  | "TU_CHOI";

export type TrangThaiDonThuoc = "CHO_DUYET" | "DA_DUYET" | "TU_CHOI";

export type DonHangDanhSach = {
  maDonHang: number;
  maKhachHang: number | null;
  tenKhachHang: string | null;
  soDienThoaiKhachHang: string | null;
  maDonThuoc: number | null;
  maNhanVienXuLy: number | null;
  tenNhanVienXuLy: string | null;
  ngayDatHang: string;
  loaiKhach: LoaiKhachHang | null;
  tongTienHang: number;
  phiGiaoHang: number | null;
  giamGia: number | null;
  tongThanhToan: number;
  phuongThucThanhToan: PhuongThucThanhToan | null;
  trangThaiThanhToan: TrangThaiThanhToan;
  trangThaiDonHang: TrangThaiDonHang;
  trangThaiKiemDuyet: TrangThaiKiemDuyetDonHang;
  coThuocKeDon: boolean;
};

export type ChiTietDonHang = {
  maChiTietDonHang: number;
  maSanPham: number;
  tenSanPham: string;
  laThuocKeDon: boolean;
  hinhAnh: string | null;
  maDonViSanPham: number;
  maDonViTinh: number;
  tenDonViTinh: string;
  kyHieu: string | null;
  soLuong: number;
  donGia: number;
  giamGia: number | null;
  thanhTien: number;
  cachTinhGia: string | null;
};

export type DonHangChiTiet = {
  maDonHang: number;
  maKhachHang: number | null;
  tenKhachHang: string | null;
  soDienThoaiKhachHang: string | null;

  maDiaChi: number | null;
  tenNguoiNhan: string | null;
  soDienThoaiNhan: string | null;
  thanhPho: string | null;
  phuongKhuVuc: string | null;
  diaChiChiTiet: string | null;

  maVoucher: number | null;
  maDonThuoc: number | null;

  maNhanVienXuLy: number | null;
  tenNhanVienXuLy: string | null;

  ngayDatHang: string;
  loaiKhach: LoaiKhachHang | null;

  tongTienHang: number;
  phiGiaoHang: number | null;
  giamGia: number | null;
  tongThanhToan: number;

  phuongThucThanhToan: PhuongThucThanhToan | null;
  trangThaiThanhToan: TrangThaiThanhToan;
  trangThaiDonHang: TrangThaiDonHang;

  trangThaiKiemDuyet: TrangThaiKiemDuyetDonHang;
  ngayKiemDuyet: string | null;
  ghiChuKiemDuyet: string | null;
  lyDoTuChoiDuyet: string | null;

  ghiChu: string | null;

  anhDonThuoc: string | null;
  trangThaiDonThuoc: TrangThaiDonThuoc | null;
  lyDoTuChoiDonThuoc: string | null;
  ghiChuDonThuoc: string | null;

  coThuocKeDon: boolean;
  danhSachChiTiet: ChiTietDonHang[];
};

export type DonHangBoLoc = {
  page: number;
  size: number;
  keyword?: string;
  trangThaiDonHang?: string;
  trangThaiThanhToan?: string;
  trangThaiKiemDuyet?: string;
};
